src/NuGet.Core/NuGet.Protocol/VisualStudio/ExtensibleSourceRepositoryProvider.cs

```diff
@@ -1,6 +1,8 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+#if IS_DESKTOP
```

Comment:
I did that first and it worked, but then moved to this. I'm open to either way.

My current thinking is that if defs are more obvious when working in Visual Studio on the code. I've seen VS have issues in the past with evaluating conditions such as the IsDesktop property we have there on items.

Also, in VS you get the switcher drop down at the top to go back and forth. With having this in the csproj you wouldn't get that.

If defs are ugly however, so I can understand wanting this in the csproj.

Thoughts?